ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/sync/BlockPropagationManagerTest.java

```diff
@@ -77,7 +82,7 @@ public static void setupSuite() {
   @Before
   public void setup() {
     blockchainUtil = BlockchainSetupUtil.forTesting();
-    blockchain = spy(blockchainUtil.getBlockchain());
+    blockchain = blockchainUtil.getBlockchain();
```

Comment:
Using a spy here causes problems because `spy` doesn't wrap the existing object, it creates a new instance.  So we wind up with two instances of `DefaultMutableBlockchain` - one inside `BlockchainSetupUtil` which isn't a spy and the one in `blockchain` which is and they wind up getting out of sync. Sadly this creates some pain in the two tests that wanted to verify how many times a block was imported but it's at least a little more controlled and we consistently use the spies.

Reply:
What if you wrap the original object in a spy inside of `BlockchainSetupUtil`?

Reply:
It's possible to do it that way but using spies is a really invasive technique - you don't get the test isolation you do from mocks but pick up the brittleness of testing specific interactions rather than results.  So I was really hesitant to embed it in such a commonly used util and encourage it to spread.